cohttp-async/src/server.ml

```diff
@@ -152,36 +143,50 @@ let respond_with_file ?flush ?headers ?(error_body=error_body_default) filename
   |Ok res -> return res
   |Error _exn -> respond_string ~status:`Not_found error_body
 
-type mode = Conduit_async.server
-
-let create_raw ?max_connections ?backlog ?buffer_age_limit ?(mode=`TCP)
-    ~on_handler_error where_to_listen handle_request =
-  Conduit_async.serve ?max_connections ?backlog
-    ?buffer_age_limit ~on_handler_error mode
-    where_to_listen (handle_client handle_request)
-  >>| fun server ->
-  { server }
-
+let reader_and_writer_of_flow flow =
+  match Conduit_async.cast flow Conduit_async.TCP.protocol,
+        Conduit_async.cast flow Conduit_async_ssl.TCP.protocol with
+  | Some flow, None ->
+    Async.return (Conduit_async.TCP.Protocol.reader flow,
+                  Conduit_async.TCP.Protocol.writer flow)
+  | None, Some flow ->
+    let { Conduit_async_ssl.reader; writer; _ } = flow in
+    Async.return (reader, writer)
+  | _ -> Conduit_async.reader_and_writer_of_flow flow
+
+let create_raw ?max_connections:_ ?backlog:_ ?buffer_age_limit:_ ~protocol ~service
```

Comment:
Similarly as in the other case, the new conduit is dropping `max_connections`. `backlog` and `buffer_age_limit`. Is this something to worry about? Is there a reason for keep it here if not?